test/common/upstream/round_robin_load_balancer_fuzz_test.cc

```diff
@@ -0,0 +1,43 @@
+#include <memory>
+
+#include "test/common/upstream/round_robin_load_balancer_fuzz.pb.validate.h"
+#include "test/common/upstream/zone_aware_load_balancer_fuzz_base.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+DEFINE_PROTO_FUZZER(const test::common::upstream::RoundRobinLoadBalancerTestCase& input) {
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  }
+
+  ZoneAwareLoadBalancerFuzzBase zone_aware_load_balancer_fuzz = ZoneAwareLoadBalancerFuzzBase(
+      input.zone_aware_load_balancer_test_case().need_local_priority_set(),
+      input.zone_aware_load_balancer_test_case().random_bytestring_for_weights());
+  zone_aware_load_balancer_fuzz.initializeLbComponents(
+      input.zone_aware_load_balancer_test_case().load_balancer_test_case());
+  zone_aware_load_balancer_fuzz.setupZoneAwareLoadBalancingSpecificLogic();
+
+  try {
+    zone_aware_load_balancer_fuzz.lb_ = std::make_unique<RoundRobinLoadBalancer>(
+        zone_aware_load_balancer_fuzz.priority_set_,
+        zone_aware_load_balancer_fuzz.local_priority_set_.get(),
+        zone_aware_load_balancer_fuzz.stats_, zone_aware_load_balancer_fuzz.runtime_,
+        zone_aware_load_balancer_fuzz.random_,
+        input.zone_aware_load_balancer_test_case().load_balancer_test_case().common_lb_config());
+  } catch (EnvoyException& e) {
+    ENVOY_LOG_MISC(debug, "EnvoyException; {}", e.what());
+    return;
+  }
+
```

Comment:
How different is this to other zone-aware LB fuzzers? Wondering if this should be the RR LB fuzzer or just a generic zone-aware one with some enum for the concrete type.

Reply:
Hmmmm, yeah the main distinction that I've noticed is the setup distinction in the unit tests for Round Robin vs. Least Request. Round Robin sets host weights, while Least request sets the amount of requests. Perhaps I could simply do an enum like you mentioned, use the same random bytestring, except for extending logic in regards to update flags, switch on the type, and either update host weights or amounts of requests.

Reply:
@asraa linking you on this - high level design comment which I like the sound of, what do you think?

Reply:
My two cents: If the fuzzer is fairly slow, having two is better. 
Actual answer: If speed weren't an issue:
How much added complexity is there in handling the setup for multiple types of zone aware fuzzers in a base zone aware fuzzer? If number of requests setup can happen in the same loop as setting host weight, for e.g., then that's not bad, and the benefit of having less overall code and just switching by enum sounds good to me! 
If you run the risk of slowing down all zone aware ones because the base has to do many time consuming operations to handle state for all of them, then I would seperate them.

Reply:
I think the state it has to set up (host weights vs. num of requests) is same. However, I think that we should switch on enum here while still having two separate fuzzers. I.e. what we have now x2 for LeastRequest, and handle all the logic for both in this ZoneAwareBase class using that enum. What do you two think?

Reply:
> I think the state it has to set up (host weights vs. num of requests) is same.

The time it takes to set up state?

> However, I think that we should switch on enum here while still having two separate fuzzers. I.e. what we have now x2 for LeastRequest, and handle all the logic for both in this ZoneAwareBase class using that enum. 

Is this what you mean: handle both in ZoneAwareFuzzBase with a boolean/enum for how hosts are set up. Then create two fuzzers that instantiate base class differently.

If the setup can be done at the same time, then the only reason to separate is logical separation and running both in parallel. Is that the justification?

Reply:
Just to document this, me and Asra discussed this in standup. The current plan is to handle both weights and num_of_requests in the zone_aware, which switches on a type enum to determine type logic. Then, there will be two separate fuzzers for both Round Robin and Least Request. The main pros me and Asra saw in this were 1. being able to reuse random_bytestring_ for both weights and # of requests on hosts, and 2. being able to run both fuzzers in parallel.